src/plots/linePlot.ts

```diff
@@ -61,6 +140,136 @@ export module Plots {
       return new Plottable.Drawers.Line(dataset);
     }
 
+    protected _extentsForProperty(property: string) {
+      let extents = super._extentsForProperty(property);
+
+      if (!this._autorangeSmooth) {
+        return extents;
+      }
+
+      if (this.autorangeMode() !== property) {
+        return extents;
+      }
+
+      if (this.autorangeMode() !== "x" && this.autorangeMode() !== "y") {
+        return extents;
+      }
+
+      let edgeIntersectionPoints = this._getEdgeIntersectionPoints();
+      let includedValues: number[];
+      if (this.autorangeMode() === "y") {
+        includedValues = edgeIntersectionPoints.left.concat(edgeIntersectionPoints.right).map((point) => point.y);
+      } else { // === "x"
+        includedValues = edgeIntersectionPoints.top.concat(edgeIntersectionPoints.bottom).map((point) => point.x);
+      }
+
+      let maxIncludedValue = Math.max.apply(this, includedValues);
+      let minIncludedValue = Math.min.apply(this, includedValues);
+
+      extents = extents.map((extent) => {
+        // When no end-points are in the viewport
+        if (extent.length === 0) {
+          return [minIncludedValue, maxIncludedValue];
+        }
+        return [Math.min(minIncludedValue, extent[0]), Math.max(maxIncludedValue, extent[1])];
+      });
+
+      return extents;
+    }
+
+    private _getEdgeIntersectionPoints(): EdgeIntersection {
+      if (!(this.y().scale instanceof QuantitativeScale && this.x().scale instanceof QuantitativeScale)) {
+        return {
+          top: [],
+          bottom: [],
+          left: [],
+          right: []
+        };
+      }
+
+      let yScale = <QuantitativeScale<number>>this.y().scale;
+      let xScale = <QuantitativeScale<any>>this.x().scale;
+
+      let intersectionPoints: EdgeIntersection = {
+        top: [],
+        bottom: [],
+        left: [],
+        right: []
+      };
+      let leftX = xScale.scale(xScale.domain()[0]);
+      let rightX = xScale.scale(xScale.domain()[1]);
+      let bottomY = yScale.scale(yScale.domain()[0]);
+      let topY = yScale.scale(yScale.domain()[1]);
+
+      this.datasets().forEach((dataset) => {
+        let data = dataset.data();
+
+        let x1: number, x2: number, y1: number, y2: number;
+        let prevX: number, prevY: number, currX: number, currY: number;
+        for (let i = 1; i < data.length; i++) {
+          prevX = currX || xScale.scale(this.x().accessor(data[i - 1], i - 1, dataset));
+          prevY = currY || yScale.scale(this.y().accessor(data[i - 1], i - 1, dataset));
+
+          currX = xScale.scale(this.x().accessor(data[i], i, dataset));
+          currY = yScale.scale(this.y().accessor(data[i], i, dataset));
+
+          // If values crossed left edge
+          if ((prevX < leftX) === (leftX <= currX)) {
```

Comment:
The point is already included by some other `includedValuesProvider`, I take it?

Also, since the line returns `false`, we avoid the division by zero, so this edge case might be okay. Maybe we should just leave it like this?